### PR TITLE
[PLGN-760] Snortlabslist - update vulns, sdk, add unit tests

### PR DESCRIPTION
**Pull request by llaszuk-r7**
### Description

Describe the proposed changes:

  - update vulnerability
  - update sdk
  - add unit tests

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/snortlabslist/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 # List third-party dependencies here, separated by newlines. 
 # All dependencies must be version-pinned, eg. requests==1.2.0
 # See: https://pip.pypa.io/en/stable/user_guide/#requirements-files
-requests==2.20.0
\ No newline at end of file
+requests==2.31.0
```
Comment (ablakley-r7) [inline plugins/snortlabslist/requirements.txt:4]: Can we remove requests entirely and rely on the SDK 2.31.0?

File: plugins/snortlabslist/unit_test/test_lookup.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        mock_response = mock_get.return_value
+        mock_response.text = "1.2.3.4\n5.6.7.8\n"
+        action = Lookup()
+        params = {"address": "9.10.11.12"}
+
+        result = action.run(params)
+
+        self.assertFalse(result["found"])
+        self.assertEqual(result["address"], "9.10.11.12")
+        self.assertEqual(result["status"], "No Error")
```
Comment (igorski-r7) [inline plugins/snortlabslist/unit_test/test_lookup.py:36]: Could you use `parameterized` to parameterise those tests? Also, that'd be awesome to move some parts to `def setUp(self) -> None` such as `action` etc 😄

File: plugins/snortlabslist/help.md [hunk 3]
```diff
@@ -47,26 +48,40 @@ Example input:
...
 
 ## Troubleshooting
-
-_This plugin does not contain any troubleshooting information._
+  
+*There is no troubleshooting for this plugin.*
 
 # Version History
 
+* 1.0.3 - Update requests to version 2.31.0
```
Comment (cmcnally-r7) [inline plugins/snortlabslist/help.md:84]:
```suggestion
* 1.0.3 - Update SDK | Update requests to version 2.31.0
```

**Review by cmcnally-r7 — APPROVED** [review]
Just a changelog update but lgtm 😃

**Review by ablakley-r7 — APPROVED** [review]

**Pull request merged**